catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/converter/impl/AbstractFeatureConverterWfs11.java

```diff
@@ -0,0 +1,215 @@
+
+  AbstractFeatureConverterWfs11(String srs) {
+    this.setSrs(srs);
+  }
+
+  AbstractFeatureConverterWfs11(MetacardMapper metacardMapper) {
+    super(metacardMapper);
+  }
+
+  protected Serializable getValueForMetacardAttribute(
+      AttributeFormat attributeFormat, HierarchicalStreamReader reader) {
+
+    Serializable ser = null;
+    switch (attributeFormat) {
+      case BOOLEAN:
+        ser = Boolean.valueOf(reader.getValue());
+        break;
+      case DOUBLE:
+        ser = Double.valueOf(reader.getValue());
+        break;
+      case FLOAT:
+        ser = Float.valueOf(reader.getValue());
+        break;
+      case INTEGER:
+        ser = Integer.valueOf(reader.getValue());
+        break;
+      case LONG:
+        ser = Long.valueOf(reader.getValue());
+        break;
+      case SHORT:
+        ser = Short.valueOf(reader.getValue());
+        break;
+      case XML:
+      case STRING:
+        ser = reader.getValue();
+        break;
+      case GEOMETRY:
+        XmlNode node = new XmlNode(reader);
+        String geometryXml = node.toString();
+        Geometry geo;
+
+        geo = readGml(geometryXml);
+
+        LOGGER.debug("coordinateOrder = {}", coordinateOrder);
+        if (GeospatialUtil.LAT_LON_ORDER.equals(coordinateOrder)) {
+          swapCoordinates(geo);
+        }
+
+        if (geo != null) {
+          WKTWriter wktWriter = new WKTWriter();
+          ser = wktWriter.write(geo);
+          LOGGER.debug("wkt = {}", ser);
+        }
+        break;
+      case BINARY:
+        try {
+          ser = reader.getValue().getBytes(UTF8_ENCODING);
+        } catch (UnsupportedEncodingException e) {
+          LOGGER.debug("Error encoding the binary value into the metacard.", e);
+        }
+        break;
+      case DATE:
+        ser = parseDateFromXml(reader);
+        break;
+      default:
+        break;
+    }
+    return ser;
+  }
+
+  private Geometry readGml(String xml) {
+    LOGGER.debug("readGml() input XML: {}", xml);
+    // Add namespace into XML for processing
+    DocumentBuilder dBuilder;
+    Document doc = null;
+    Object gml = null;
+    InputStream xmlIs;
+
+    // Check if GML 3.1.1 namespace exist on XML chunk
+    try {
+      dBuilder = XML_UTILS.getSecureDocumentBuilder(false);
+      InputSource is = new InputSource();
+      is.setCharacterStream(new StringReader(xml));
+      doc = dBuilder.parse(is);
+    } catch (ParserConfigurationException | SAXException | IOException e) {
+      LOGGER.debug(XML_PARSE_FAILURE, e);
+    }
+
+    if (null != doc) {
+      String[] namePrefix = doc.getDocumentElement().getNodeName().split(":");
+      String prefix = "";
+      if (namePrefix.length < 2) {
+        LOGGER.debug("Incoming XML has no GML prefix");
+      } else {
+        prefix = ":" + namePrefix[0];
+      }
+
+      String xmlNs = doc.getDocumentElement().getAttribute("xmlns" + prefix);
+      if (xmlNs.equals(Wfs11Constants.GML_3_1_1_NAMESPACE)) {
```

Comment:
Can xmlNs be null here?

Reply:
it could be, in theory. I'll swap the order so that we never have a NPE